contrib/opencensus-ext-azure/tests/test_azure_log_exporter.py

```diff
@@ -0,0 +1,130 @@
+# limitations under the License.
+
+import logging
+import mock
+import os
+import shutil
+import unittest
+
+from opencensus.ext.azure import log_exporter
+
+TEST_FOLDER = os.path.abspath('.test.logs')
+
+
+def setUpModule():
+    os.makedirs(TEST_FOLDER)
+
+
+def tearDownModule():
+    shutil.rmtree(TEST_FOLDER)
+
+
+def throw(exc_type, *args, **kwargs):
+    def func(*_args, **_kwargs):
+        raise exc_type(*args, **kwargs)
+    return func
+
+
+class CustomLogHandler(log_exporter.BaseLogHandler):
+    def __init__(self, max_batch_size, callback):
+        self.export_interval = 1
+        self.max_batch_size = max_batch_size
+        self.callback = callback
+        super(CustomLogHandler, self).__init__()
+
+    def export(self, batch):
+        return self.callback(batch)
+
+
+class TestBaseLogHandler(unittest.TestCase):
+    def test_basic(self):
+        logger = logging.getLogger(self.id())
+        handler = CustomLogHandler(10, lambda batch: None)
+        logger.addHandler(handler)
+        logger.warning('TEST')
+        handler.flush()
+        logger.warning('TEST')
+        logger.removeHandler(handler)
+        handler.close()
+
+    def test_export_exception(self):
+        logger = logging.getLogger(self.id())
+        handler = CustomLogHandler(1, throw(Exception))
+        logger.addHandler(handler)
+        logger.warning('TEST')
+        logger.removeHandler(handler)
+        handler.flush()
+        handler.close()
+
+
+class TestAzureLogHandler(unittest.TestCase):
+    def test_ctor(self):
+        from opencensus.ext.azure.common import Options
+        instrumentation_key = Options._default.instrumentation_key
+        Options._default.instrumentation_key = None
+        self.assertRaises(ValueError, lambda: log_exporter.AzureLogHandler())
+        Options._default.instrumentation_key = instrumentation_key
+
+    @mock.patch('requests.post', return_value=mock.Mock())
+    def test_exception(self, requests_mock):
+        logger = logging.getLogger(self.id())
+        handler = log_exporter.AzureLogHandler(
+            instrumentation_key='12345678-1234-5678-abcd-12345678abcd',
+            storage_path=os.path.join(TEST_FOLDER, self.id()),
+        )
+        logger.addHandler(handler)
+        try:
+            return 1 / 0  # generate a ZeroDivisionError
+        except Exception:
+            logger.exception('Captured an exception.')
+        handler.close()
+
+    @mock.patch('requests.post', return_value=mock.Mock())
+    def test_export_empty(self, request_mock):
+        handler = log_exporter.AzureLogHandler(
+            instrumentation_key='12345678-1234-5678-abcd-12345678abcd',
+            storage_path=os.path.join(TEST_FOLDER, self.id()),
+        )
+        handler._export([])
+        self.assertEqual(len(os.listdir(handler.storage.path)), 0)
+        handler.close()
+
+    @mock.patch('opencensus.ext.azure.log_exporter.AzureLogHandler.log_record_to_envelope')  # noqa: E501
```

Comment:
Nit, feel free to ignore if you disagree:

```suggestion
    @mock.patch('opencensus.ext.azure.log_exporter'
                '.AzureLogHandler.log_record_to_envelope')

```

Reply:
Taken, thanks.